src/include/connection.h

```diff
@@ -573,6 +574,7 @@ struct __wt_connection_impl {
     bool mmap_all; /* use mmap for all I/O on data files */
     int page_size; /* OS page size for mmap alignment */
 
+    /* Access to these fields is protected by the debug_log_retention_lock. */
     WT_LSN *debug_ckpt;      /* Debug mode checkpoint LSNs. */
```

Comment:
It might be worth doing a CTRL+F for any of these variables to see if they're used outside of locks.

Reply:
Confirmed. The only use outside of a lock is freeing of `debug_ckpt` in `__wt_connection_destroy`

Reply:
Heh, does that need a lock?